test/kotlin/com/example/people_managment_system/controller/SystemControllerTest.kt

```diff
@@ -0,0 +1,308 @@
+package com.example.people_managment_system.controller
+
+import com.example.people_managment_system.dtos.EmployeeDto
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.MediaType
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.test.web.servlet.*
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ContextConfiguration
+internal class SystemControllerTest @Autowired constructor (
+    val mockMvc: MockMvc,
+    val objectMapper: ObjectMapper
+    ) {
+
+    val baseUrl = "/api/system"
+
+    @Nested
+    @DisplayName("getEmployee()")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+
+    inner class GetEmployee {

```

Comment:
This is one way of doing it but there is another way and this is basically mocking the SystemService class.

You can check this guide out in how to mock with mockito: https://www.baeldung.com/kotlin/mockito